extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -125,60 +122,78 @@ export class AmpAdXOriginIframeHandler {
       this.sendEmbedInfo_(this.baseInstance_.isInViewport());
     }));
 
+    // Iframe.onload normally called by the Ad after full load.
+    const iframeLoadPromise = loadPromise(this.iframe).then(() => {
+      // Wait just a little to allow messages to arrive.
+      return timer.promise(10);
+    });
     if (this.baseInstance_.emitLifecycleEvent) {
       // Only set up a load listener if we know that we can send lifecycle
       // messages.
-      loadPromise(this.iframe).then(() => {
+      iframeLoadPromise.then(() => {
         this.baseInstance_.emitLifecycleEvent('xDomIframeLoaded');
       });
     }
 
-    // Install API that listens to ad response
-    if (this.baseInstance_.config
-        && this.baseInstance_.config.renderStartImplemented) {
-      // If support render-start, create a race between render-start no-content
-      this.adResponsePromise_ = listenForOncePromise(this.iframe,
+    // Calculate render-start and no-content signal. These signals are mutually
+    // exlcusive. Whichever arrives first wins.
+    let renderStartResolve;
+    const renderStartPromise = new Promise(resolve => {
+      renderStartResolve = resolve;
+    });
+    let noContentResolve;
+    const noContentPromise = new Promise(resolve => {
+      noContentResolve = resolve;
+    });
+    listenForOncePromise(this.iframe,
         ['render-start', 'no-content'], true).then(info => {
           const data = info.data;
           if (data.type == 'render-start') {
             this.renderStart_(info);
+            renderStartResolve();
           } else {
             this.noContent_();
+            noContentResolve();
           }
         });
-    } else {
-      // If NOT support render-start, listen to bootstrap-loaded no-content
-      // respectively
-      this.adResponsePromise_ = listenForOncePromise(this.iframe,
-        'bootstrap-loaded', true);
```

Comment:
I'm thinking about bringing it back in the similar fashion we had before. It's not a very decisive points since we are strongly moving toward having render-start in all of the important cases.